### ISSUE TITLE
Android OS 11+ Launching Broken

### ISSUE THREAD

**Issue by zach-morris** [zach-morris/plugin.program.iagl#255]
Android OS versions 11+ have changed their permissions to be even more restrictive than they were in the past.

When launching a game, the launch will fail, and the log will show something to the effect of:
```
ActivityTaskManager: Permission Denial: package=com.android.shell does not belong to uid=xxx
```
It is the same issue other frontend launchers had with the OS, ex:
https://github.com/mmatyas/pegasus-frontend/issues/766

However, in IAGL's case, I'm restricted to whatever I can do only within Python.  At the moment, there's no workaround that I'm aware of.  Something would have to change either in the OS or in Kodi to get launching working again for Android 11+

**Comment by dataghost2014** [comment]
Binaries need editing, and launch from neutral location. Getting closer with changes being made, will share when more progress is made. Cheers.

**Comment by chrisism** [comment]
@dataghost2014 Might extending the StartAndroidActivity method help in this case? It looks like that builtin method is working with launching android apps, but it lacks enough properties needed to do correct app launching in launcher addons like IAGL.
See the request on kodi forum for more details: https://forum.kodi.tv/showthread.php?tid=367878

I'll follow this issue too if any updates available.

**Comment by zach-morris** [comment]
Some work is being done in Kodi, which looks promising.  Unsure how it will interact with permissions though:
https://github.com/xbmc/xbmc/pull/21652

**Comment by zach-morris** [comment]
Fixed in 3.0.7 -  the wiki will have to be updated but there are some caveats. Main one being that the files need to be downloaded to a shared directory (one that both apps can see given their permissions). The wizard is updated to account for this. Let me know if you run into any issues.
